### DISCUSSION TITLE
custom_path_generators doesn't work with custom polymorphic types

### DISCUSSION THREAD

**Discussion by kskrlinnorth2** [spatie/laravel-medialibrary#3034] [Ideas]
I am using custom names for my morphed models, for example `user` instead of `App\Models\User` and similar generated via `enforceMorphMap` method on `Illuminate\Database\Eloquent\Relations\Relation` class, however `getPathGeneratorClass` method can't work with such approach since `is_a` expects full model namespaces so I can't use different custom path generators for each model.

I fixed it with this block of code; could you update package with same or similar solution? Solution is to check if media model type exists in custom morphed list and then getting original model name via existing alias, otherwise just get the model type as in original code.
My installed package version is 10.3.6. (installed via this command: `composer require "spatie/laravel-medialibrary:^10.0.0"`)

```
protected static function getPathGeneratorClass(Media $media)
{
    $defaultPathGeneratorClass = config('media-library.path_generator');

    foreach (config('media-library.custom_path_generators', []) as $modelClass => $customPathGeneratorClass) {
        $modelType = array_key_exists($media->model_type, Relation::$morphMap)
            ? Relation::getMorphedModel($media->model_type)
            : $media->model_type;

        if (is_a($modelType, $modelClass, true)) {
            return $customPathGeneratorClass;
        }
    }
 
     return $defaultPathGeneratorClass;
}
```

**Comment by freekmurze** [comment]
Could you send a PR for this? Be sure to also included a test to ensure it works.

**Reply by kskrlinnorth2** [reply]
I created PR for this, but I don't really know how to run those checks to run tests.. Also I need to create and commit Unit tests for this particular part of code first?

**Comment by aprokopenko** [comment]
I have a similar issue and create the PR with the hotfix:
https://github.com/spatie/laravel-medialibrary/pull/3191

**Reply by aprokopenko** [reply]
@freekmurze , you mentioned a "test" above. Do you mean, that you want a PHPUnit test to cover this particular fix? If yes, should I add test in some existing TestCase file or create a new one?

**Reply by aprokopenko** [reply]
Do you have any instructions on how you run tests for the particular package? (I have my project installed and tested this code with it, but I don't think I can run tests with my project for particular package from vendors)
